src/index.ts

```diff
@@ -65,11 +84,17 @@ export class TypeScriptPlugin {
             delete require.cache[module]
           })
         }
+        this.watchProgress = this.watchProgress ?? this.utils.progress.create()
       },
-      'after:invoke:local:invoke': () => {
+      'after:invoke:local:invoke': async () => {
         if (this.options.watch) {
+          if (this.watchProgress) {
+            this.watchProgress.update('Watching for TypeScript changes')
+          }
           this.watchFunction()
-          this.serverless.cli.log('Waiting for changes...')
+          return new Promise(() => {
```

Comment:
More accurate way would be to await on watchFunction, and fix `watchFunction` so it returns properly resolved promise.

Also it'll be great to have that patch in current major of a plugin (so best if it's proposed with other PR)

Reply:
Yes, that's what I tried initially. But I think it's more complex than that, because `watchFunction()` calls `watchFiles()` that watches multiple files in parallel, and recursively re-spawns the CLI command in callbacks of each file changed. It seems like it isn't as simple as awaiting everything, given the command calls itself kind of recursively.

In other words, it seems to me like fixing this would require a more substantial refactoring/re-architecting of the way the plugin works, and that would be a bit too much for that pull request. If you manage to do that on master I can rebase on it of course, but I'd rather not attempt that here for lack of time, complexity, and also because this isn't really related to the redesign.

Reply:
Ok, I'll give it a try against master and let you know